src/Distributed/multi_architectures.jl

```diff
@@ -1,35 +1,62 @@
 using Oceananigans.Architectures
-import Oceananigans.Architectures: device_event
+using Oceananigans.Grids: topology, validate_tupled_argument, with_arch
 
-using Oceananigans.Grids: topology, validate_tupled_argument
+import Oceananigans.Architectures: device, device_event, arch_array
+import Oceananigans.Grids: zeros
 
-struct MultiCPU{G, R, I, ρ, C, γ} <: AbstractCPUArchitecture
-    distributed_grid :: G
+struct MultiArch{A, R, I, ρ, C, γ} <: AbstractMultiArchitecture
+          child_arch :: A
```

Comment:
```suggestion
          child_architecture :: A
```

?

Probably best to remain consistent throughout regarding nomenclature...